File: letter_capitalize/letter_capitalize.rb
# alternative to .titleize or .titlecase
def title_case(sentence)
  words = sentence.split(' ')
  words.each do |word|
    word[0] = word[0].capitalize
  end
  words = words.join(' ')
end

puts title_case("hello world")
puts title_case("i ran there")

File: letter_capitalize/letter_capitalize.js
function capitalizeFirst(word) {
  return word.charAt(0).toUpperCase() + word.slice(1);
}

function titleCase(sentence) {
  var words = sentence.split(" ");
  var result = "";
  for (var i = 0; i < words.length; i++) {
    result = result.concat(capitalizeFirst(words[i]));
    if (i != words.length - 1) {
      result = result.concat(" ");
    }
  }
  return result;
}

console.log(titleCase("i ran there"));